src/cargo/util/command_prelude.rs

```diff
@@ -331,7 +331,16 @@ pub trait CommandExt: Sized {
     }
 
     fn arg_quiet(self) -> Self {
+        let unsupported_silent_arg = || {
+            let msg = "there is no `--silent` flag, use `--quiet` to suppress output";
+            let value_parser = UnknownArgumentValueParser::suggest(msg);
```

Comment:
I feel like the default message is sufficient. We probably don't need/want to customize everything every time.

`tip: a similar argument exists: '--quiet'`

```suggestion
            let value_parser = UnknownArgumentValueParser::suggest_arg("--quiet");
```

Reply:
I'm also hoping we can make the rust-native completion engine complete `suggest_arg` flags with what should be used instead.

Reply:
Good point!